Tests/test_file_avif.py

```diff
@@ -0,0 +1,683 @@
+class TestFileAvif:
+    def test_version(self):
+        _avif.AvifCodecVersions()
+        assert re.search(r"\d+\.\d+\.\d+$", features.version_module("avif"))
+
+    def test_read(self):
+        """
+        Can we read an AVIF file without error?
+        Does it have the bits we expect?
+        """
+
+        with Image.open("Tests/images/avif/hopper.avif") as image:
+            assert image.mode == "RGB"
+            assert image.size == (128, 128)
+            assert image.format == "AVIF"
+            assert image.get_format_mimetype() == "image/avif"
+            image.load()
+            image.getdata()
+
+            # generated with:
+            # avifdec hopper.avif hopper_avif_write.png
+            assert_image_similar_tofile(
+                image, "Tests/images/avif/hopper_avif_write.png", 12.0
+            )
+
+    def _roundtrip(self, tmp_path, mode, epsilon, args={}):
+        temp_file = str(tmp_path / "temp.avif")
+
+        hopper(mode).save(temp_file, **args)
+        with Image.open(temp_file) as image:
+            assert image.mode == "RGB"
+            assert image.size == (128, 128)
+            assert image.format == "AVIF"
+            image.load()
+            image.getdata()
+
+            if mode == "RGB":
+                # avifdec hopper.avif avif/hopper_avif_write.png
+                assert_image_similar_tofile(
+                    image, "Tests/images/avif/hopper_avif_write.png", 12.0
+                )
+
+            # This test asserts that the images are similar. If the average pixel
+            # difference between the two images is less than the epsilon value,
+            # then we're going to accept that it's a reasonable lossy version of
+            # the image.
+            target = hopper(mode)
+            if mode != "RGB":
+                target = target.convert("RGB")
+            assert_image_similar(image, target, epsilon)
+
+    def test_write_rgb(self, tmp_path):
+        """
+        Can we write a RGB mode file to avif without error?
+        Does it have the bits we expect?
+        """
+
+        self._roundtrip(tmp_path, "RGB", 12.5)
+
+    def test_AvifEncoder_with_invalid_args(self):
+        """
+        Calling encoder functions with no arguments should result in an error.
+        """
+        with pytest.raises(TypeError):
+            _avif.AvifEncoder()
+
+    def test_AvifDecoder_with_invalid_args(self):
+        """
+        Calling decoder functions with no arguments should result in an error.
+        """
+        with pytest.raises(TypeError):
+            _avif.AvifDecoder()
+
+    def test_no_resource_warning(self, tmp_path):
+        with Image.open(TEST_AVIF_FILE) as image:
+            temp_file = str(tmp_path / "temp.avif")
+            pytest.warns(None, image.save, temp_file)
+
+    def test_file_pointer_could_be_reused(self):
+        with open(TEST_AVIF_FILE, "rb") as blob:
+            Image.open(blob).load()
+            Image.open(blob).load()
```

Comment:
```suggestion
            with Image.open(blob) as im:
                im.load()
            with Image.open(blob) as im:
                im.load()
```
Using context managers, as per https://pillow.readthedocs.io/en/stable/releasenotes/7.0.0.html#image-del